src/expression/node/AccessorNode.js

```diff
@@ -133,10 +134,13 @@ export const createAccessorNode = /* #__PURE__ */ factory(name, dependencies, ({
 
     /**
      * Create a clone of this node, a shallow copy
+     * @param {MetaOptions} object with additional options for cloning this node
      * @return {AccessorNode}
      */
-    clone () {
-      return new AccessorNode(this.object, this.index)
+    clone (meta = {}) {
```

Comment:
The default value for meta, an empty object `{}`, does not correspond with the official TypeScript definition where `sources` is a require property.

I think it is best to have `sources` are required property. I would also like to get rid of mutating the input `meta` via `meta.sources = ...`, so I think we can rewrite the clone methods to something like: 

```js
    /**
     * Create a clone of this node, a shallow copy
     * @param {MetaOptions} [meta] object with additional options for cloning this node
     * @return {AccessorNode}
     */
    clone (meta) {
      const cloned = new AccessorNode(this.object, this.index, meta ?? { sources: this.sources })
      return cloned
    }
```

Having to construct a default meta object via `{ sources: this.sources }` is a bit odd. Would it make sense to just keep the meta object as it is instead of destructuring it in the `Node` constructor, and having to re-construct it in the methods `clone` before we can pass it to another constructor? I.e we could think about:

```js
  const defaultMetaOptions = {
    sources = []
  }

  class Node {
    /**
     * @constructor Node
     * A generic node, the parent of other AST nodes
     * @param {MetaOptions} [meta] object with additional options for building this node
     */
    constructor (meta = defaultMetaOptions) {
      this.meta = meta
    }
    // ...
  }
```